lib/CL/devices/pthread/pthread.c

```diff
@@ -117,6 +113,10 @@ struct data {
   mem_regions_management* mem_regions;
 #endif
 
+  /* Lock for command list related operations */
+  pthread_mutex_t cq_lock __attribute__ ((aligned (CACHELINE_SIZE)));
```

Comment:
Not all, i'd guess maybe 3/4 of it. Rest is likely from `get_wg_index_range()` tuning.

I'd say the most critical (WRT alignment) are `struct scheduler_data`, which is shared across all driver threads, and `struct pool_thread_data` which is allocated one per thread - but it was allocated via calloc() so every driver thread got something not cacheline aligned. Both these structs are heavily used by driver threads.